include/xtypes/DynamicData.hpp

```diff
@@ -227,6 +227,216 @@ class ReadableDynamicDataRef
+        {}
+
+        Iterator& operator = (
+                const Iterator& other)
+        {
+            data_ = other.data_;
+            index_ = other.index_;
+            return *this;
+        }
+
+        bool operator == (
+                const Iterator& other) const
+        {
+            return other.data_ == data_ && other.index_ == index_;
+        }
+
+        bool operator != (
+                const Iterator& other) const
+        {
+            return other.data_ != data_ || other.index_ != index_;
+        }
+
+        ReadableDynamicDataRef operator * () const
+        {
+            const CollectionType& collection = static_cast<const CollectionType&>(type_);
+            return ReadableDynamicDataRef(collection.content_type(), collection.get_instance_at(data_, index_));
+        }
+
+        Iterator& operator ++ ()
+        {
+            ++index_;
+            return *this;
+        }
+
+        Iterator operator ++ (
+                int)
+        {
+            Iterator prev = *this;
+            ++index_;
+            return prev;
+        }
+
+    protected:
+
+        friend class ReadableDynamicDataRef;
+
+        Iterator(
+                const ReadableDynamicDataRef& ref,
+                bool end)
+            : type_(ref.type_)
+            , data_(ref.instance_)
+            , index_(end ? ref.size() : 0)
+        {
+        }
+
+        const DynamicType& type_;
+        uint8_t* data_;
+        size_t index_;
+    };
+
+    /// \brief Returns the initial iterator of a collection dynamic data.
+    /// \pre The DynamicData must represent a CollectionType.
+    /// \returns The initial iterator.
+    Iterator begin() const
+    {
+        assert(type_.is_collection_type());
+        return Iterator(*this);
+    }
+
+    /// \brief Returns the final iterator of a collection dynamic data.
+    /// \pre The DynamicData must represent a CollectionType.
+    /// \returns The final iterator.
+    Iterator end() const
+    {
+        assert(type_.is_collection_type());
+        return Iterator(*this, true);
+    }
+
+    class MemberPair
+    {
+    public:
+        MemberPair(
+                const Member& member,
+                uint8_t* data)
+            : member_(member)
+            , data_(data)
+        {}
+
+        const Member& member() const
+        {
+            return member_;
+        }
+
+        ReadableDynamicDataRef data() const
+        {
+            return ReadableDynamicDataRef(member_.type(), data_);
+        }
+
+        // Shortcuts
+        TypeKind kind() const
+        {
+            return member_.type().kind();
+        }
+
+        const DynamicType& type() const
+        {
+            return member_.type();
+        }
+
+        const std::string& name() const
+        {
+            return member_.name();
+        }
+
+        template<typename T, class = PrimitiveOrString<T>>
+        T& value() const
+        {
+            return data().value<T>();
```

Comment:
With the cast operator, returning the data will be the same as the following functions. Maybe no longer necessary?

Reply:
Remove most of shortcut methods.